diff --git a/libraries/extractor/src/main/java/androidx/media3/extractor/ts/PesReader.java b/libraries/extractor/src/main/java/androidx/media3/extractor/ts/PesReader.java
index 580b66c5e5..1397dc91ad 100644
--- a/libraries/extractor/src/main/java/androidx/media3/extractor/ts/PesReader.java
+++ b/libraries/extractor/src/main/java/androidx/media3/extractor/ts/PesReader.java
@@ -132,7 +132,10 @@ public final class PesReader implements TsPayloadReader {
           break;
         case STATE_READING_HEADER:
           if (continueRead(data, pesScratch.data, HEADER_SIZE)) {
-            setState(parseHeader() ? STATE_READING_HEADER_EXTENSION : STATE_FINDING_HEADER);
+            setState(
+                parseHeader(/* logUnexpectedPrefix= */ true)
+                    ? STATE_READING_HEADER_EXTENSION
+                    : STATE_FINDING_HEADER);
           }
           break;
         case STATE_READING_HEADER_EXTENSION:
@@ -177,7 +180,7 @@ public final class PesReader implements TsPayloadReader {
   public boolean canConsumeSynthesizedEmptyPusi(boolean isModeHls) {
     // Only call parseHeader if isModeHls is true and can parse header as some HLS streams may
     // contain packages.
-    boolean headerParsed = !isModeHls || parseHeader();
+    boolean headerParsed = !isModeHls || parseHeader(/* logUnexpectedPrefix= */ false);
     // Either the PES packet does not have a length field and the body is being read (where the end
     // of the stream also signals the end of the body), or we are waiting for the next PES packet to
     // start and the end of the stream means there won't be any more data.
@@ -219,13 +222,15 @@ public final class PesReader implements TsPayloadReader {
     return bytesRead == targetLength;
   }
 
-  private boolean parseHeader() {
+  private boolean parseHeader(boolean logUnexpectedPrefix) {
     // Note: see ISO/IEC 13818-1, section 2.4.3.6 for detailed information on the format of
     // the header.
     pesScratch.setPosition(0);
     int startCodePrefix = pesScratch.readBits(24);
     if (startCodePrefix != 0x000001) {
-      Log.w(TAG, "Unexpected start code prefix: " + startCodePrefix);
+      if (logUnexpectedPrefix) {
+        Log.w(TAG, "Unexpected start code prefix: " + startCodePrefix);
+      }
       payloadSize = C.LENGTH_UNSET;
       return false;
     }
